File: nim/armstrong-numbers/armstrong_numbers.nim
import math
import sequtils
  
proc isArmstrongNumber*(number: int): bool =
  var digits: seq[int]
  var current = number

  while current != 0:
    digits.add(current mod 10)
    current = current div 10

  number == digits.mapIt(it ^ digits.len).sum()
    

File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Linq;
using System.Collections.Generic;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        var digits = new List<int>();
        var current = number;

        while (current > 0)
        {   
            digits.Add(current % 10);
            current = current / 10;
        }

        var sum = digits.Select(n => Math.Pow(n, digits.Count)).Sum();
        
        return sum == number;
    }
}
